src/Dotnet.Script.Core/Templates/program.publish.template

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Scripting.Hosting;
+using Microsoft.CodeAnalysis.Scripting.Hosting;
+using System.Threading.Tasks;
+using static System.Console;
+
+namespace dotnetPublishCode
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            try
+            {
+                var globals = new CommandLineScriptGlobals(Console.Out, CSharpObjectFormatter.Instance);
+                foreach (var arg in args)
+                    globals.Args.Add(arg);
+
+                var factoryMethod = typeof(scriptAssembly).GetMethod("<Factory>");
+                if (factoryMethod == null) throw new Exception("couldn't find factory method to initiate script");
+
+                // todo: not sure what the second parameter is, using 'null' for now
+                var invokeTask = factoryMethod.Invoke(null, new object[] { new object[] { globals, null } }) as Task<int>;
+                var invokeResult = invokeTask.Result;
+                if (invokeResult != 0) WritePrettyError($"Error result: '{invokeResult}'");
+            }
+            // todo: not getting the full script stack trace
+            catch (AggregateException ex)
```

Comment:
I think you want something like this

```
                if (e is AggregateException aggregateEx)
                {
                    e = aggregateEx.Flatten().InnerException;
                }
```

Reply:
wouldn't i need to still loop through the flattened "InnerExceptions" of a flattened aggregate exception typically?  I think you're right though in this case there will only ever be 1 inner exception in that aggregate exception.

Reply:
I think on these code paths there can only ever be one